### PR TITLE
chore(deps): upgrade `fvm` and `cid`

### LINKED ISSUE #5014: Upgrade `fvm` and `cid`
# Summary

# Completion Criteria

- [ ]

# Additional Links & Resources

### PR DESCRIPTION
**Pull request by hanabi1224**
## Summary of changes

(Based on #5013)

Changes introduced in this pull request:

- upgrade `fvm` and `cid`

## Reference issue to close (if applicable)

Closes #5014

## Change checklist

- [x] I have performed a self-review of my own code,
- [x] I have made corresponding changes to the documentation. All new code adheres to the team's [documentation standards](https://github.com/ChainSafe/forest/wiki/Documentation-practices),
- [x] I have added tests that prove my fix is effective or that my feature works (if possible),
- [ ] I have made sure the [CHANGELOG][1] is up-to-date. All user-facing changes should be reflected in this document.

[1]: https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md

### CODE REVIEW

File: src/utils/net.rs [hunk 1]
```diff
@@ -42,7 +41,8 @@ pub async fn download_ipfs_file_trustlessly(
     {
         let mut reader = reader(url.as_str()).await?.compat();
         let mut writer = futures::io::BufWriter::new(async_fs::File::create(&tmp).await?);
-        rs_car_ipfs::single_file::read_single_file_seek(&mut reader, &mut writer, Some(cid))
+        let cid_v10 = crate::utils::cid::cid_11_to_10(cid);
+        rs_car_ipfs::single_file::read_single_file_seek(&mut reader, &mut writer, Some(&cid_v10))
```
Comment (hanabi1224) [inline src/utils/net.rs:45]: `cid_0_10` is needed until https://github.com/dapplion/rs-car/pull/13 is merged and released.

File: src/utils/multihash.rs [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+pub mod prelude {
+    pub use super::MultihashCodeLegacy;
+    pub use multihash_codetable::{Code as MultihashCode, MultihashDigest as _};
+}
+
+use multihash_derive::MultihashDigest;
+
+#[derive(Clone, Copy, Debug, Eq, MultihashDigest, PartialEq)]
+#[mh(alloc_size = 64)]
+pub enum MultihashCodeLegacy {
```
Comment (ruseinov) [inline src/utils/multihash.rs:19]: Wow, that's quite a change on their part.

Comment (ruseinov) [reply src/utils/multihash.rs:19]: Do we then need all of this code, or should we instead introduce a helper method to wrap this: https://github.com/multiformats/rust-multihash/blob/master/examples/identity.rs

Comment (hanabi1224) [reply src/utils/multihash.rs]: We do need the code. e.g. `CarBlock::validate`

```rust
pub fn validate(&self) -> anyhow::Result<()> {
        let actual = {
            let code = MultihashCode::try_from(self.cid.hash().code())?;
            Cid::new_v1(self.cid.codec(), code.digest(&self.data))
        };
        anyhow::ensure!(
            actual == self.cid,
            "CID/Block mismatch for block {}, actual: {actual}",
            self.cid
        );
        Ok(())
    }
```

File: deny.toml [hunk 3]
```diff
@@ -73,7 +73,6 @@ ignore = [
   # Unmaintained crates that fvm2 requires, and will not change for
   # compatability/consensus reasons - see
   # https://github.com/filecoin-project/ref-fvm/issues/1843
-  "RUSTSEC-2020-0168", # mach is unmaintained
   "RUSTSEC-2022-0061", # parity-wasm is deprecated
   "RUSTSEC-2024-0370", # proc-macro-error is unmaintained
```
Comment (hanabi1224) [inline deny.toml:77]: `RUSTSEC-2024-0370` will be fixed once we deprecate `cid@0.10`

File: Cargo.toml [hunk 4]
```diff
@@ -94,7 +95,7 @@ humantime = "2"
 indexmap = { version = "2", features = ["serde"] }
 indicatif = { version = "0.17", features = ["tokio"] }
 integer-encoding = "4.0"
-ipld-core = { version = "0.4", features = ["serde", "arb"] }
+ipld-core = { version = "0.4", features = ['arb', 'serde'] }
```
Comment (elmattic) [inline Cargo.toml:98]: any reason to use simple quotes here?

**Review by elmattic — APPROVED** [review]

File: Cargo.toml (hunk 4 above)
Comment (hanabi1224) [inline Cargo.toml:98]:
```suggestion
ipld-core = { version = "0.4", features = ["serde", "arb"] }
```

Comment (hanabi1224) [reply Cargo.toml]: Reverted.

Comment (hanabi1224) [reply Cargo.toml]: Please re-approve @elmattic

**Review by elmattic — APPROVED** [review]

File: Cargo.toml (hunk 4 above)
Comment (elmattic) [reply Cargo.toml]: Sorry, done.

**Pull request merged**